scipy/optimize/_simulated_annealing.py

```diff
@@ -0,0 +1,557 @@
+                if self.not_improved_idx >= self.not_improved_max_idx:
+                    if j == 0 or self.state.current_energy < self.emin:
+                        self.emin = self.state.current_energy
+                        self.xmin = np.copy(self.state.current_location)
+        # End of MarkovChain loop
+
+    def local_search(self):
+        # Decision making for performing a local search
+        # based on Markov chain results
+        # If energy has been improved or no improvement since too long,
+        # performing a local search with the best Markov chain location
+        if self.state_improved:
+            # Global energy has improved, let's see if LS improved further
+            e, x = self.obj_fun_wrapper.local_search(self.state.xbest)
+            if e < self.state.ebest:
+                self.not_improved_idx = 0
+                self.state.ebest = e
+                self.state.xbest = np.copy(x)
+                self.state.current_energy = e
+                self.state.current_location = np.copy(x)
+                return
+        # Check probability of a need to perform a LS even if no improvment
+        # (Dual annealing principle)
+        do_ls = False
+        if self.K < 90 * len(self.state.current_location):
+            pls = np.exp(self.K * (self.state.ebest - self.state.current_energy
+                                   ) / self.temperature_step)
+            if pls >= self._rand_state.random_sample():
+                do_ls = True
+        # Global energy not improved, let's see what LS gives
+        # on the best Markov chain location
+        if self.not_improved_idx >= self.not_improved_max_idx:
+            do_ls = True
+        if do_ls:
+            e, x = self.obj_fun_wrapper.local_search(self.xmin)
+            self.xmin = np.copy(x)
+            self.emin = e
+            self.not_improved_idx = 0
+            self.not_improved_max_idx = self.state.current_location.size
+            if e < self.state.ebest:
+                self.state.ebest = self.emin
+                self.state.xbest = np.copy(self.xmin)
+                self.state.current_energy = e
+                self.state.current_location = np.copy(x)
+
+
+class ObjectiveFunWrapper(object):
+    """
+    Class used to wrap around the objective function in order to apply local
+    search.
+    Default local minimizer is SciPy minimizer L-BFGS-B
+    """
+    
+    MAXITER_RATIO = 6
+    MAXITER_MIN = 100
+    MAXITER_MAX = 1000
+
+    def __init__(self, bounds, func, **kwargs):
+        self.func = func
+        self.nb_fun_call = 0
+        self.kwargs = kwargs
+        self.minimizer = minimize
+        self.fun_args = ()
+        bounds_list = list(zip(*bounds))
+        self.lower = np.array(bounds_list[0])
+        self.upper = np.array(bounds_list[1])
+        if 'args' in self.kwargs:
+            self.fun_args = self.kwargs.get('args')
+        
+    def set_default_minimizer(self):
```

Comment:
missing a docstring

Reply:
is it necessary to have this method?  Can't you set up the object with these as the default, and overwrite them if the options are in `kwargs`?